resources/keda.yaml

```diff
@@ -9247,3 +9247,36 @@ spec:
     namespace: keda
   version: v1beta1
   versionPriority: 100
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: keda-operator-metrics
+  namespace: keda
+  labels:
+    app.kubernetes.io/part-of: keda-operator
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/part-of: keda-operator
```

Comment:
I'm proposing this selector to address both Services.

Reply:
As long as it's easy to use the labels in prometheus to look at metrics for one or the other. For example, it's important that we be able to see metrics just from the API server pod(s) and not have them intertwined with the keda operator metrics. As long as there are enough labels in prometheus to tease out the metrics from two pods, then I'm fine with that. (Sorry that I don't have enough prometheus knowledge to know)

Reply:
Correct. To my understanding, the proposed label's configuration in pods allows the ServiceMonitor to match the metrics endpoint for the Prometheus CR. I think a Prometheus query could be adjusted narrow the data and isolate the metrics.

@joelsmith could you add more color with a specific example so that I can test it in a dev cluster?

Reply:
I just tried this out on a dev cluster and it looks like we will need 2 servicemonitor objects because they have different port names.  One uses `metrics` for its port name and one uses `http` and `metrics`, but the `metrics` port 9022 is [deprecated](https://keda.sh/docs/2.10/operate/prometheus/) and so we need to use `http` on it. The keda metrics are all available from the operator's `metrics` port, but both the API server and the operator both expose runtime metrics which are useful for debugging. The operator exposes them on the `metrics` port and the API server exposes them on the `http` port. Here is the config that worked:

```
---
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: keda-operator-metrics
  namespace: keda
  labels:
    app.kubernetes.io/name: keda-operator
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: keda-operator
  endpoints:
    - port: metrics
      interval: 60s
  namespaceSelector:
    any: false
---
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: keda-metrics-apiserver
  namespace: keda
  labels:
    app.kubernetes.io/name: keda-metrics-apiserver
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: keda-metrics-apiserver
  endpoints:
    - port: http
      interval: 60s
  namespaceSelector:
    any: false
```

Reply:
@joelsmith  Thanks for validating this; I updated the PR with the corrections.

Reply:
> the metrics port 9022 is [deprecated](https://keda.sh/docs/2.10/operate/prometheus/) and so we need to use http on it.

WRT to the deprecation, should we create a card to remove the deprecated port and name it accordingly, i.e. `http` to `metrics` ?

Reply:
I'm not sure that it would be appropriate to rename the port since it has more endpoints than just `/metrics`. It also has the API server endpoint. It just happens to serve both endpoints from the same port.

Reply:
Understood.